lib/Doctrine/ODM/MongoDB/Mapping/Annotations/DiscriminatorField.php

```diff
@@ -4,13 +4,23 @@
 
 namespace Doctrine\ODM\MongoDB\Mapping\Annotations;
 
-use Doctrine\Common\Annotations\Annotation;
+use Attribute;
+use Doctrine\Common\Annotations\Annotation\NamedArgumentConstructor;
 
 /**
  * Specify a field name to store a discriminator value
  *
  * @Annotation
+ * @NamedArgumentConstructor
  */
-final class DiscriminatorField extends Annotation
+#[Attribute(Attribute::TARGET_CLASS | Attribute::TARGET_PROPERTY)]
```

Comment:
```suggestion
#[Attribute(Attribute::TARGET_CLASS)]
```
Based on https://www.doctrine-project.org/projects/doctrine-mongodb-odm/en/2.2/reference/annotations-reference.html#discriminatorfield looks like it's only used at class level (same for `DefaultDiscriminatorValue` and `DiscriminatorMap`).